application/comit_node/src/comit_i_routes.rs

```diff
@@ -0,0 +1,98 @@
+use comit_i::Asset;
+use http::{uri::PathAndQuery, Response, StatusCode};
+use mime_guess::Mime;
+use std::{
+    borrow::Cow,
+    error::Error as StdError,
+    fmt::{self, Display},
+    str::FromStr,
+};
+use warp::{filters::BoxedFilter, path::Tail, Filter, Rejection, Reply};
+
+#[derive(Copy, Clone, Debug)]
+enum Error {
+    IndexHtmlMissing,
+    PathConversionFail,
+}
+
+#[derive(Serialize)]
+struct ErrorMessage {
+    code: u16,
+    message: String,
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.description())
+    }
+}
+
+impl From<Error> for Rejection {
+    fn from(err: Error) -> Self {
+        warp::reject::custom(err)
+    }
+}
+
+impl StdError for Error {
+    fn description(&self) -> &str {
+        match self {
+            Error::IndexHtmlMissing => "index.html file not found",
+            Error::PathConversionFail => "Conversion of the path to PathAndQuery failed",
+        }
+    }
+
+    fn cause(&self) -> Option<&dyn StdError> {
+        None
+    }
+}
+
+pub fn create() -> BoxedFilter<(impl Reply,)> {
+    warp::any()
+        .and(warp::path::tail())
+        .and_then(serve)
+        .recover(unpack_problem)
+        .boxed()
+}
+
+fn serve(path: Tail) -> Result<impl Reply, Rejection> {
+    let path = path.as_str();
+    let path_and_query = PathAndQuery::from_str(path).map_err(|e| {
+        error!("Could not convert path {} to PathAndQuery: {:?}", path, e);
+        Error::PathConversionFail
+    })?;
+    let path = path_and_query.path();
+
+    let mut mime = mime_guess::guess_mime_type_opt(path)
```

Comment:
:wrench: Maybe there is a way to avoid the mutability + double parsing here:

1. `let asset: Option<Cow<'static, [u8]>> = Asset::get(path);`
2. `map` asset to a response + guess the mimetype ([`guess_mime_type`](https://docs.rs/mime_guess/2.0.0-alpha.6/mime_guess/fn.guess_mime_type.html) seems to more appropriate here instead of defaulting to `text/html`)
3. `unwrap_or_else` the option to serve the `index.html` asset by constructing a `Response` that statically sets the `content-type` header to "text/html" (atm you are parsing first and a `to_string` afterwards.

You can fetch the `index.html` asset upfront, since nothing works without that one anyway :)

```rust
let index = Asset::get("index.html").ok_or_else(|| Error::IndexHtmlMissing)?
```

Reply:
This causes a bit of duplication in constructing the `Response`, however, I'd argue that it is overall preferable since you keep the special-case of serving the `index.html` in case the asset is not found separate.

You could even extract it in its own function to not clutter up this one.